payments-core/src/main/java/com/stripe/android/paymentsheet/model/PaymentOptionFactory.kt

```diff
@@ -2,42 +2,31 @@ package com.stripe.android.paymentsheet.model
 
 import android.content.res.Resources
 import com.stripe.android.R
-import com.stripe.android.model.CardBrand
-import com.stripe.android.model.PaymentMethod
 
 internal class PaymentOptionFactory(
     private val resources: Resources
 ) {
-    fun create(selection: PaymentSelection): PaymentOption? {
+    fun create(selection: PaymentSelection): PaymentOption {
         return when (selection) {
             PaymentSelection.GooglePay -> {
-                // TODO(mshafrir-stripe): update values
                 PaymentOption(
                     drawableResourceId = R.drawable.stripe_google_pay_mark,
-                    label = "Google Pay"
+                    label = resources.getString(R.string.google_pay)
                 )
             }
             is PaymentSelection.Saved -> {
-                when (selection.paymentMethod.type) {
-                    PaymentMethod.Type.Card -> {
-                        selection.paymentMethod.card?.let { card ->
-                            PaymentOption(
-                                drawableResourceId = getIcon(card.brand),
-                                label = createCardLabel(card.last4)
-                            )
-                        }
-                    }
-                    else -> {
-                        // TODO(mshafrir-stripe): handle other types
-                        null
-                    }
-                }
+                PaymentOption(
+                    drawableResourceId = selection.paymentMethod.getSavedPaymentMethodIcon() ?: 0,
```

Comment:
nit: seems `getSavedPaymentMethodIcon` could be a extension of `PaymentSelection` directly if there is always going to be a icon attached to a selection

Reply:
Yes, the logic could live in `PaymentSelection` too, but I think this is a good place because they are only needed when we're creating the `PaymentOption`. `PaymentSelection` only holds values.